src/update_handler.c

```diff
@@ -1034,6 +1080,157 @@ static gboolean img_to_fs_handler(RaucImage *image, RaucSlot *dest_slot, const g
 	return res;
 }
 
+static gboolean img_to_boot_emmc_handler(RaucImage *image, RaucSlot *dest_slot, const gchar *hook_name, GError **error) {
+
+	gboolean res = FALSE;
+	int out_fd;
+	gint ret = 0;
+	gint part_active;
+	gint part_active_after;
+	GOutputStream *outstream = NULL;
+	GError *ierror = NULL;
+	RaucSlot *part_slot = NULL;
+
+	/* run slot pre install hook if enabled */
+	if (hook_name && image->hooks.pre_install) {
+		res = run_slot_hook(
+				hook_name,
+				R_SLOT_HOOK_PRE_INSTALL,
+				NULL,
+				dest_slot,
+				&ierror);
+		if (!res) {
+			g_propagate_error(error, ierror);
+			goto out;
+		}
+	}
+
+	/* read active boot partition from ext_csd */
+	ret = r_emmc_read_bootpart(dest_slot->device, &part_active, &ierror);
+	if (!ret) {
+		g_propagate_error(error, ierror);
+		res = FALSE;
+		goto out;
+	}
+	g_message("Found active eMMC boot partition %sboot%d",
+			dest_slot->device, part_active);
+
+	/* create a temporary RaucSlot with the actual partition as device */
+	part_slot = g_new0(RaucSlot, 1);
+
+	/* determine active boot partition and set the other to be written */
+	part_slot->device = g_strdup_printf(
+			"%sboot%d",
+			dest_slot->device,
+			INACTIVE_BOOT_PARTITION(part_active));
+
+	/* disable read-only on determined eMMC boot partition */
+	g_message("Disabling read-only mode of slot device partition %s",
+			part_slot->device);
+	res = r_emmc_force_part_rw(part_slot->device, &ierror);
+	if (!res) {
+		g_propagate_error(error, ierror);
+		goto out;
+	}
+
+	/* clear block device */
+	g_message("Clearing slot device %s", part_slot->device);
+	res = clear_slot(part_slot, &ierror);
+	if (!res) {
+		g_propagate_error(error, ierror);
+		goto out;
+	}
+
+	/* open */
+	g_message("Opening slot device %s", part_slot->device);
+	outstream = open_slot_device(part_slot, &out_fd, &ierror);
+	if (outstream == NULL) {
+		g_propagate_error(error, ierror);
+		res = FALSE;
+		goto out;
+	}
+
+	/* copy */
+	g_message("Copying image to slot device partition %s",
+			part_slot->device);
+	res = copy_raw_image(image, outstream, &ierror);
+	if (!res) {
+		g_propagate_error(error, ierror);
+		goto out;
+	}
+
+	/* reenable read-only on determined eMMC boot partition */
+	g_message("Reenabling read-only mode of slot device partition %s",
+			part_slot->device);
+	res = r_emmc_force_part_ro(part_slot->device, &ierror);
+	if (!res) {
+		g_propagate_error(error, ierror);
+		goto out;
+	}
+
+	/* toggle active boot partition in ext_csd register
+	 *
+	 * Do this explicitly on determined boot partition to force the kernel
+	 * to switch to the partition.
+	 */
+	g_message("Toggling active eMMC boot partition %sboot%d -> %s",
+			dest_slot->device, part_active, part_slot->device);
+
+	res = r_emmc_write_bootpart(
+			part_slot->device,
+			INACTIVE_BOOT_PARTITION(part_active),
+			&ierror);
+	if (!res) {
+		g_propagate_error(error, ierror);
+		goto out;
+	}
+
+	/* sanity check: read active boot partition from ext_csd
```

Comment:
s/read/readd/

Reply:
What does this mean?

Reply:
Skip that :D seems I confused while scrolling through the code and assumed this to be a re-add from 'adding again' but its indeed a simple 'read' ;)